agents.py

```diff
@@ -350,7 +350,70 @@ def mark(self, start:float, target_value:float):
         # Value is too high: return None
         return None
 
+class PiecewiseLinearAgent(Agent):
 
+    """
+    A PiecewiseLinearAgent is an Agent with a finite number of desired intervals, all of which have linear value-density (1).
+    
+    >>> a = PiecewiseLinearAgent([(x1,x2,a,b)]) #y=ax+b
+
```

Comment:
Please add a "__repr__" function, like the other classes in this file. Then, make sure the doctest is correct.